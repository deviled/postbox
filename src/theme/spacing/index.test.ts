import { values, spacing } from '.';

describe('spacing', () => {
  it('should match snapshot', () => {
    expect(values).toMatchSnapshot();
  });

  it('should return correct spacing', () => {
    expect(spacing(12)).toBe('12px');
  });

  it('should return correct py and px spacing', () => {
    expect(spacing(12, 24)).toBe('12px 24px');
  });

  it('should return correct pt, px and pb spacing', () => {
    expect(spacing(12, 16, 24)).toBe('12px 16px 24px');
  });

  it('should return correct pt, pr, pb, pl spacing', () => {
    expect(spacing(12, 16, 20, 24)).toBe('12px 16px 20px 24px');
  });

  it('should throw error if more than 4 arguments provided', () => {
    const throwable = () => spacing(12, 16, 20, 24, 32);

    expect(throwable).toThrow(Error);
    expect(throwable).toThrow('Max 4 arguments theme.spacing(12, 16, 20, 24, 32)');
  });
});
