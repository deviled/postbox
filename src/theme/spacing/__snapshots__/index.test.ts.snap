// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spacing should match snapshot 1`] = `
Object {
  "0": "0px",
  "12": "12px",
  "16": "16px",
  "2": "2px",
  "20": "20px",
  "24": "24px",
  "32": "32px",
  "4": "4px",
  "8": "8px",
}
`;
